src/langchain_google_alloydb_pg/utils/pgvector_migrator.py

```diff
@@ -0,0 +1,440 @@
+            Optional.
+        destination_table (str): The name of the table to insert the data in.
+            Optional. defaults to collection_name.
+        use_json_metadata (bool): An option to keep the PGVector metadata as json in the AlloyDB table.
+            Default: False. Optional.
+        delete_pg_collection (bool): An option to delete the original data upon migration.
+            Default: False. Optional.
+        insert_batch_size (int): Number of rows to insert at once in the table.
+            Default: 1000.
+    """
+    if not use_json_metadata and not metadata_columns:
+        raise ValueError(
+            "Please specify the columns for the new table. "
+            "To store data in JSON format, set use_json_metadata=True when calling this method."
+        )
+
+    if not destination_table:
+        warnings.warn(
+            f"Destination table not set. Destination table would default to {collection_name}. "
+            "Please make sure that there is an existing table with the same name."
+        )
+        destination_table = collection_name
+
+    # Extract data from the collection and batch insert into the new table
+    collection_data = _aextract_pgvector_collection(engine, collection_name)
+    data_batches = _batch_data(collection_data, insert_batch_size)
+
+    if metadata_columns:
+        vector_store = await AsyncAlloyDBVectorStore.create(
+            engine=engine,
+            embedding_service=embeddings_service,
+            table_name=destination_table,
+            metadata_columns=metadata_columns,
+        )
+    else:
+        vector_store = await AsyncAlloyDBVectorStore.create(
+            engine=engine,
+            embedding_service=embeddings_service,
+            table_name=destination_table,
+        )
+
+    tasks = [
+        asyncio.create_task(
+            _ainsert_single_batch(
+                engine,
+                vector_store,
+                data=batch_data,
+            )
+        )
+        async for batch_data in data_batches
+    ]
+    await asyncio.gather(*tasks)
+
+    # Get row count in PGVector collection
+    uuid = await _aget_collection_uuid(engine, collection_name)
+    query = f"SELECT COUNT(*) FROM {EMBEDDINGS_TABLE} WHERE collection_id='{uuid}'"
+    async with engine._pool.connect() as conn:
+        result = await conn.execute(text(query))
+        result_map = result.mappings()
+        collection_data_len = result_map.fetchone()
+    if collection_data_len is None:
+        raise ValueError(f"Collection, {collection_name} contains no elements.")
+
+    # Validate data migration
+    query = f"SELECT COUNT(*) FROM {destination_table}"
+    async with engine._pool.connect() as conn:
+        result = await conn.execute(text(query))
+        result_map = result.mappings()
+        table_size = result_map.fetchone()
+    if not table_size:
+        raise ValueError(f"Table: {destination_table} does not exist.")
```

Comment:
Is this the right error to surface? We will probably hit this error via the vector store